src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -1122,38 +1125,168 @@ class spell_pri_power_word_shield : public SpellScriptLoader
+        }
+
+        void HandleEffectDummy(SpellEffIndex effIndex)
+        {
+            SpellEffectInfo const* effectInfo = GetEffectInfo(effIndex);
+            if (!effectInfo)
+                return;
+
+            Unit* target = GetExplTargetUnit();
+            if (!target)
+                return;
+
+            Unit* caster = GetCaster();
+            CastPrayerOfMendingAura(caster, target, effectInfo->BasePoints);
+        }
+
+        void Register() override
+        {
+            OnEffectHit += SpellEffectFn(spell_pri_prayer_of_mending_SpellScript::HandleEffectDummy, EFFECT_0, SPELL_EFFECT_DUMMY);
+        }
+    };
+
+    SpellScript* GetSpellScript() const override
+    {
+        return new spell_pri_prayer_of_mending_SpellScript();
+    }
+};
+
+// 41635 - Prayer of Mending (Aura) - SPELL_PRIEST_PRAYER_OF_MENDING_AURA
+class spell_pri_prayer_of_mending_aura : public SpellScriptLoader
 {
     public:
-        spell_pri_prayer_of_mending_heal() : SpellScriptLoader("spell_pri_prayer_of_mending_heal") { }
+        spell_pri_prayer_of_mending_aura() : SpellScriptLoader("spell_pri_prayer_of_mending_aura") { }
 
-        class spell_pri_prayer_of_mending_heal_SpellScript : public SpellScript
+        class spell_pri_prayer_of_mending_aura_AuraScript : public AuraScript
         {
-            PrepareSpellScript(spell_pri_prayer_of_mending_heal_SpellScript);
+            PrepareAuraScript(spell_pri_prayer_of_mending_aura_AuraScript);
+
+            bool Validate(SpellInfo const* /*spellInfo*/) override
+            {
+                return ValidateSpellInfo({ SPELL_PRIEST_PRAYER_OF_MENDING_HEAL });
+            }
 
-            void HandleHeal(SpellEffIndex /*effIndex*/)
+            void HandleHeal(ProcEventInfo & /*eventInfo*/)
             {
-                if (Unit* caster = GetOriginalCaster())
+                // Caster: the player(priest) that casters the prayer of mending
+                // Owner: the player that currently has the prayer of mending aura on him
+                Unit* owner = GetUnitOwner();
+                Aura* aura = owner->GetAura(SPELL_PRIEST_PRAYER_OF_MENDING_AURA);
+                if (Unit* caster = GetCaster())
                 {
-                    if (AuraEffect* aurEff = caster->GetAuraEffect(SPELL_PRIEST_T9_HEALING_2P, EFFECT_0))
-                    {
-                        int32 heal = GetHitHeal();
-                        AddPct(heal, aurEff->GetAmount());
-                        SetHitHeal(heal);
-                    }
+                    // Cast the spell to heal the owner
+                    caster->CastSpell(owner, SPELL_PRIEST_PRAYER_OF_MENDING_HEAL);
+
+                    // Only cast jump if stack is higher than 0
+                    if (aura->GetStackAmount() > 1)
+                        owner->CastSpell(owner, SPELL_PRIEST_PRAYER_OF_MENDING_JUMP, true, nullptr, nullptr, caster->GetGUID());
+                    else
+                        aura->Remove();
```

Comment:
```suggestion
                    if (aura->GetStackAmount() > 1)
                        owner->CastSpell(owner, SPELL_PRIEST_PRAYER_OF_MENDING_JUMP, true, nullptr, nullptr, caster->GetGUID());

                    Remove();
```

You always want to remove the aura from original unit

Reply:
Jump spell needs the aura to check the stack amount.